task/net/src/server.rs

```diff
@@ -577,18 +710,26 @@ where
             return Err(ClientError::AccessViolation.fail());
         }
 
+        // Convert from absolute VID to an index in our VLAN array
         #[cfg(feature = "vlan")]
-        let vlan_index = {
-            // Convert from absolute VID to an index in our VLAN array
-            if !VLAN_RANGE.contains(&metadata.vid) {
-                return Err(ClientError::BadMessageContents.fail());
-            }
-            usize::from(metadata.vid - VLAN_RANGE.start)
+        let Some(vlan_index) = VLANS.iter().position(|v| v.vid == metadata.vid) else {
+            return Err(ClientError::BadMessageContents.fail());
         };
+
         #[cfg(not(feature = "vlan"))]
         let vlan_index = 0;
 
         let vlan = &mut self.vlan_state[vlan_index];
+
+        // Refuse to send messages directed to an untrusted VLAN, silently
+        // dropping them.
+        let now = userlib::sys_get_timer().now;
+        let trust = vlan.check_trust(now)
+            | generated::SOCKET_ACCEPT_UNTRUSTED[socket_index];
+        if !trust {
+            return Ok(());
```

Comment:
This should hit a counter.

Reply:
I added a ringbuf message and made it a counted ringbuf, so we'll now get counters for free!